tests/integration/suite/scheduler/quorum/notls.go

```diff
@@ -0,0 +1,144 @@
+		scheduler.WithInitialClusterPorts(fp.Port(t, 0), fp.Port(t, 1), fp.Port(t, 2)),
+	}
+
+	clientPorts := []string{
+		"scheduler0=" + strconv.Itoa(fp.Port(t, 3)),
+		"scheduler1=" + strconv.Itoa(fp.Port(t, 4)),
+		"scheduler2=" + strconv.Itoa(fp.Port(t, 5)),
+	}
+	n.schedulers = []*scheduler.Scheduler{
+		scheduler.New(t, append(opts, scheduler.WithID("scheduler0"), scheduler.WithEtcdClientPorts(clientPorts))...),
+		scheduler.New(t, append(opts, scheduler.WithID("scheduler1"), scheduler.WithEtcdClientPorts(clientPorts))...),
+		scheduler.New(t, append(opts, scheduler.WithID("scheduler2"), scheduler.WithEtcdClientPorts(clientPorts))...),
+	}
+
+	fp.Free(t)
+	return []framework.Option{
+		framework.WithProcesses(n.schedulers[0], n.schedulers[1], n.schedulers[2]),
+	}
+}
+
+func (n *notls) Run(t *testing.T, ctx context.Context) {
+	n.schedulers[0].WaitUntilRunning(t, ctx)
+	n.schedulers[1].WaitUntilRunning(t, ctx)
+	n.schedulers[2].WaitUntilRunning(t, ctx)
+
+	// Schedule job to random scheduler instance
+	chosenScheduler := n.schedulers[rand.Intn(3)]
+
+	host := chosenScheduler.Address()
+	conn, err := grpc.DialContext(ctx, host, grpc.WithBlock(), grpc.WithReturnConnectionError(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	require.NoError(t, err)
+
+	t.Cleanup(func() { require.NoError(t, conn.Close()) })
+
+	client := schedulerv1pb.NewSchedulerClient(conn)
+
+	jobName := "appID||testJob"
+	req := &schedulerv1pb.ScheduleJobRequest{
+		Job: &runtimev1pb.Job{
+			Name:     jobName,
+			Schedule: "@every 1s",
+		},
+		Namespace: "default",
+		Metadata:  nil,
+	}
+
+	_, err = client.ScheduleJob(ctx, req)
+	require.NoError(t, err)
+
+	// Keep this, I believe the record is added to the db at trigger time, so I need the sleep for the db check to pass
+	time.Sleep(2 * time.Second) // allow time for the record to be added to the db
+
+	chosenSchedulerPort := chosenScheduler.EtcdClientPort()
+	require.NotEmptyf(t, chosenSchedulerPort, "chosenSchedulerPort should not be empty")
+
+	etcdCmdOutput := executeEtcdCmd(t, chosenSchedulerPort)
+	checkCmdOutputForAppID(t, etcdCmdOutput)
+
+	// ensure data exists on ALL schedulers
+	for i := 0; i < 3; i++ {
+		diffScheduler := n.schedulers[i]
+
+		diffSchedulerPort := diffScheduler.EtcdClientPort()
+		require.NotEmptyf(t, diffSchedulerPort, "diffSchedulerPort should not be empty")
+
+		diffEtcdCmdOutput := executeEtcdCmd(t, diffSchedulerPort)
+		checkCmdOutputForAppID(t, diffEtcdCmdOutput)
+	}
+}
+
+// TODO: Cassie - rm this and switch to the get func once the get functionality checks from the db
+// as of now, the get func only checks in memory, so the data wont be there on another scheduler instance
+func executeEtcdCmd(t *testing.T, port string) string {
+	cmd := exec.Command("etcdctl", "get", "", "--prefix", fmt.Sprintf("--endpoints=localhost:%s", port))
```

Comment:
Please don't require etcdctl, you can use the etcd's go library instead.

Reply:
Oh, let me look into that